src/ui/public/utils/aggressive_parse.js

```diff
@@ -0,0 +1,27 @@
+import { isString, startsWith } from 'lodash';
+
+/**
+ * Serializes the given object into a JSON string
+ *
+ * All properties that begin with $ throughout the entire object are omitted.
+ * If a custom JSON serializer function is passed, then the given object is
+ * passed through it before being re-stringified with the native stringify.
+ *
+ * The space argument is passed unaltered to the native stringify.
+ */
+export function toJson(object, jsonFn, space) {
```

Comment:
I don't understand the benefit of the `jsonFn` argument. If it's for performance reasons as the comment below implies it seems far less performant to encode, then decode, and then encode again rather than just encoding.